src/eyreish/into_diagnostic.rs

```diff
@@ -1,12 +1,20 @@
-use thiserror::Error;
+use std::{error::Error, fmt::Display};
 
 use crate::{Diagnostic, Report};
 
 /// Convenience [`Diagnostic`] that can be used as an "anonymous" wrapper for
 /// Errors. This is intended to be paired with [`IntoDiagnostic`].
-#[derive(Debug, Error)]
-#[error(transparent)]
+#[derive(Debug)]
 pub(crate) struct DiagnosticError(pub(crate) Box<dyn std::error::Error + Send + Sync + 'static>);
+
+impl Display for DiagnosticError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let msg = &self.0;
+        write!(f, "{msg}")
+    }
+}
+impl Error for DiagnosticError {}
```

Comment:
`error(transparent)` forwards the `source` method

Reply:
Ach! Good catch! I think my tests must have missed that by not using an error with a source... I'll fix that up now!

Reply:
Should be fixed now!